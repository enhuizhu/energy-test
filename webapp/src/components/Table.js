import React from 'react';

export default function(props) {
  return (
    <table>
      <tbody>
        <tr>
          {
            props.headers.map((v, k) => {
              return (<th key={k}>{v.label}</th>);
            })
          }
        </tr>
        {
          props.rowData.map((v, k) => {
            return (<tr key={k}>
              {
                 props.headers.map((header, i) => {
                  return <td key={i}>{v[header.key]}</td>;
                })
              }
            </tr>)
          })
        }
      </tbody>
    </table>
  );
}