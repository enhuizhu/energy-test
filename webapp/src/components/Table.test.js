import React from 'react';
import Table from './Table';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';

describe('table component', () => {
  const headers = [{label:'a', key: 'a'}, {label:'b', key: 'b'}, {label:'c', key: 'c'}];
  const rowData = [{a: 1, b: 2, c:3}];
  
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Table 
          headers={headers}
          rowData={rowData}
        >
        </Table>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('testing content inside', () => {
    const table = shallow(<Table 
      headers={headers}
      rowData={rowData}
    ></Table>);
    expect(table.html()).toEqual('<table><tbody><tr><th>a</th><th>b</th><th>c</th></tr><tr><td>1</td><td>2</td><td>3</td></tr></tbody></table>');
  });
});