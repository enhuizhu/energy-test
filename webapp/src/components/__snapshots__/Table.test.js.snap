// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`table component renders correctly 1`] = `
<table>
  <tbody>
    <tr>
      <th>
        a
      </th>
      <th>
        b
      </th>
      <th>
        c
      </th>
    </tr>
    <tr>
      <td>
        1
      </td>
      <td>
        2
      </td>
      <td>
        3
      </td>
    </tr>
  </tbody>
</table>
`;
